spinnaker/spinnaker_system/aws_smoke_test.py

```diff
@@ -102,13 +96,17 @@ def __init__(self, bindings, agent=None):
       agent: [GateAgent] The agent for invoking the test operations on Gate.
     """
     super(AwsSmokeTestScenario, self).__init__(bindings, agent)
-
     bindings = self.bindings
-    bindings['TEST_APP_COMPONENT_NAME'] = (
-        '{app}-{stack}-{detail}'.format(
-            app=bindings['TEST_APP'],
-            stack=bindings['TEST_STACK'],
-            detail=bindings['TEST_COMPONENT_DETAIL']))
+
+    optional_stack = ('-{stack}'.format(stack=bindings['TEST_STACK'])
+                      if bindings['TEST_STACK']
+                      else '')
+
+    self.__lb_detail = 'lb'
+    self.__lb_name = '{app}{optional_stack}-{detail}'.format(
```

Comment:
Pretty sure if you omit stack, you're left with two `-` characters. This should reflect that